### ISSUE TITLE
Failure to connect via SSH to CentOS 7 64 bit from Arch Linux host

### ISSUE THREAD

**Issue by PlasmaPower** [mitchellh/vagrant#7657]
### Vagrant version
Vagrant 1.8.5

### Host operating system

Arch Linux

### Guest operating system

CentOS 7 64 bit (Ubuntu Trusty 64 bit and others works fine)

### Vagrantfile

https://github.com/borgbackup/borg/blob/master/Vagrantfile

Most relevant section:

```ruby
  config.vm.define "centos7_64" do |b|
    b.vm.box = "centos/7"
    b.vm.provider :virtualbox do |v|
      v.memory = 768
    end
    b.vm.provision "install system packages", :type => :shell, :inline => packages_redhatted
    b.vm.provision "install pyenv", :type => :shell, :privileged => false, :inline => install_pyenv("centos7_64")
    b.vm.provision "install pythons", :type => :shell, :privileged => false, :inline => install_pythons("centos7_64")
    b.vm.provision "build env", :type => :shell, :privileged => false, :inline => build_pyenv_venv("centos7_64")
    b.vm.provision "install borg", :type => :shell, :privileged => false, :inline => install_borg("centos7_64")
    b.vm.provision "run tests", :type => :shell, :privileged => false, :inline => run_tests("centos7_64")
  end
```

### Debug output

https://gist.github.com/b61e0d85addb33a14598e5c253dec768

In my system log this is spammed if I have ssh-agent running (the crash occurs without it, but this error doesn't appear in my system log):

```
ssh-agent[456]: error: Unknown message 1
```

### Expected behavior
Vagrant should be able to connect to the guest via SSH.

### Actual behavior
It fails.

### Steps to reproduce
1. Get a fresh Arch installation
2. `pacman -S base-devel virtualbox virtualbox-host-modules-arch vagrant git rsync`
3. If you're in a chroot like I am, load virtualbox modules on the true host
4. git clone https://github.com/borgbackup/borg.git && cd borg
5. vagrant up centos7_64

### References
- borgbackup/borg#1384 - previously I thought this was a Vagrantfile issue

**Comment by oleyka** [comment]
I am getting a similar error in OS X 10.11.5 trying to log into a brand new CentOS7 box by puppetlabs in vagrant 1.8.5 (downgrading to 1.8.1 fixes the issue).

```
@:~/tmp$ vagrant init puppetlabs/centos-7.2-64-puppet
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.
@:~/tmp$ vagrant up
Bringing machine 'default' up with 'vmware_fusion' provider...
==> default: Removing hosts
==> default: No machine id, nothing removed from /etc/hosts
==> default: Cloning VMware VM: 'puppetlabs/centos-7.2-64-puppet'. This can take some time...
==> default: Checking if box 'puppetlabs/centos-7.2-64-puppet' is up to date...
==> default: Skipping vmnet device verification, verify_vmnet is set to false.
==> default: Preparing network adapters...
==> default: Checking for host entries
==> default: Fixed port collision for 22 => 2222. Now on port 2200.
==> default: Starting the VMware VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 172.16.64.132:22
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection refused. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...

```

**Comment by PlasmaPower** [comment]
Sounds like it's not just me. I'll try to bisect it.

**Comment by PlasmaPower** [comment]
Looks like this has been fixed on master, and will be fixed by 1.8.6.

**Comment by PlasmaPower** [comment]
Duplicate of #7610, fixed by a6760dd8e7743e048cb2f38c474e05889356e8ac.

**Comment by travistrle** [comment]
This issue still happens to me as today. I'm using vagrant 1.8.5

**Comment by PlasmaPower** [comment]
Try a newer version of Vagrant. I think it's been fixed since 1.8.6.

**Comment by amitmawkin** [comment]
It is not fixed in 1.9.4 too

centos-7.2$ vagrant init centos/7; vagrant up --provider virtualbox
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...
    default: Warning: Authentication failure. Retrying...

**Comment by amitmawkin** [comment]
removed .vagrant folder and it ran file on re initialization
